Sources/BSON/ByteBuffer+BSON.swift

```diff
@@ -10,14 +10,27 @@ extension ByteBuffer {
 
     /// Attempts to read null terminated UTF-8 string from ByteBuffer starting at the readerIndex
     internal mutating func readCString() throws -> String {
+        let string = try self.getCString(at: self.readerIndex)
+        self.moveReaderIndex(forwardBy: string.utf8.count + 1)
+        return string
+    }
+
+    /// Attempts to read null terminated UTF-8 string from ByteBuffer starting at the offset
+    internal func getCString(at offset: Int) throws -> String {
+        let key = try getBSONKey(at: offset).dropLast()
+        guard let string = String(bytes: key, encoding: .utf8) else {
+            throw BSONError.InternalError(message: "Failed to decode BSONKey as UTF8: \(key)")
+        }
+        return string
+    }
+
+    /// Returns the C String key including the null byte, for ease of comparison and byte counting
+    internal func getBSONKey(at offset: Int) throws -> [UInt8] {
```

Comment:
I think this name is confusing because C strings aren't just used for BSON keys. can we come up with a name that better clarifies what exactly this method does? for example, "getCStringBytes" ? I also don't see this method used outside of `getCString`, where will we be using it otherwise?